Sparkle/SUBinaryDeltaCommon.m

```diff
@@ -155,12 +198,41 @@ static void _hashOfFile(unsigned char* hash, FTSENT *ent)
     return @(hexHash);
 }
 
-void removeTree(NSString *path)
+extern NSString *hashOfTree(NSString *path)
 {
-    [[NSFileManager defaultManager] removeItemAtPath:path error:nil];
+    return hashOfTreeWithVersion(path, LATEST_DELTA_DIFF_MAJOR_VERSION);
 }
 
-void copyTree(NSString *source, NSString *dest)
+BOOL removeTree(NSString *path)
 {
-    [[NSFileManager defaultManager] copyItemAtPath:source toPath:dest error:nil];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    // Don't use fileExistsForPath: because it will try to follow symbolic links
+    if (![fileManager attributesOfItemAtPath:path error:nil]) {
+        return YES;
+    }
+    return [fileManager removeItemAtPath:path error:nil];
+}
+
+BOOL copyTree(NSString *source, NSString *dest)
+{
+    return [[NSFileManager defaultManager] copyItemAtPath:source toPath:dest error:nil];
+}
+
+BOOL modifyPermissions(NSString *path, mode_t desiredPermissions)
+{
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSDictionary *attributes = [fileManager attributesOfItemAtPath:path error:nil];
+    if (!attributes) {
+        return NO;
+    }
+    NSNumber *permissions = attributes[NSFilePosixPermissions];
+    if (!permissions) {
+        return NO;
+    }
+    mode_t newMode = ([permissions unsignedShortValue] & ~PERMISSION_FLAGS) | desiredPermissions;
```

Comment:
Not sure I understand. Permission modes are replaced completely with what they are in the newer file, rather than being added/removed.